### ISSUE TITLE
No log file produced after a failure

### ISSUE THREAD

**Issue by kinow** [BSC-ES/autosubmit#705]
In GitLab by @acarreri on Jun 3, 2021, 14:12

Hi @eferre1 , @mcastril ,  (sorry, again not sure to tag the correct people)

The first SIM job of my experiment, `a3qa` failed (received the notification and confirmed by the GUI) but no log file has been saved in the `/esarchive/autosubmit/a3qa/tmp/LOG_a3qa/` folder. But it is in MN4.

I don't think it's a normal behaviour. Could you have a look at it please?

Thanks

Aude

**Comment by kinow** [comment]
In GitLab by @acarreri on Jun 30, 2021, 14:33

Hi guys,

I'm sorry, I re-opening this issue, because it happened again with another experiment, `a3rp`.

So indeed, there are "special" experiments because ocean-only with some errors that don't make the SIM job fail during the execution but wait for the wallclock to fail. 

However, I could expect the log file to be saved in esarchive after the wallclock is reached, right? But it's not saved on esarchive. And actually, there is only a `a3rp_19590101_fc00_2_SIM.cmd.err` in mn4 (`/gpfs/scratch/bsc32/bsc32283/a3rp/LOG_a3rp`).

I don't understand well what happens, but I think something's wrong with the log files in this case. And I don't think it's happening without the wrappers (not totally certain though).

Could you have a look at it please?

Many thanks

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Jun 30, 2021, 14:37

Hello,

We had issues this morning (noticed during the hands on) regarding to esarchive so I think what happened is that the log retrial mechanism expired the maximum retrials because it was failing for a long period of time

**Comment by kinow** [comment]
In GitLab by @acarreri on Jun 30, 2021, 14:51

Well, if it's because of that, exactly the same problem with esarchive happened for the previous experiment mentioned in the issue.

Which could be considered as strange random errors, because it's really the same kind of "special" experiments, with wrappers and in both cases, no log files are produced for the corresponding SIM job, no even in mn4 (cf this [comment](https://earth.bsc.es/gitlab/es/autosubmit/-/issues/705#note_137485) for the previous experiment in this issue).

**Comment by kinow** [comment]
In GitLab by @mcastril on Jun 30, 2021, 16:56

@dbeltrankyl can you confirm that we retrieve the error log files when the wrapper is canceled by AS?

**Comment by kinow** [comment]
In GitLab by @mcastril on Jul 5, 2021, 15:30

@dbeltrankyl confirms that the fact that it was canceled due to wallclock should not have any impact.

**Comment by kinow** [comment]
In GitLab by @mcastril on Jul 12, 2021, 15:45

@acarreri we are talking about this problem right now.

You said that the first chunk in the wrapper was failing, but that there is only an `a3rp_19590101_fc00_2_SIM.cmd.err` file. Wasn't any logfile for the first chunk? Or is the 2nd the one that failed in the second case?

**Comment by kinow** [comment]
In GitLab by @acarreri on Jul 12, 2021, 17:08

Hi @mcastril ,

Sorry, I probably missed the "right now"!

For `a3rp`, the first SIM ran correctly. But not the second SIM, the failure was related to the surface_restoring files. It's an error that doesn't make the job failed, but just waiting to the wallclock limit. 

And for the failure of this second SIM, I never got the log file.

**Comment by kinow** [comment]
In GitLab by @acarreri on Aug 11, 2021, 14:17

Hi all,

I think that now, everything works fine. At least, after having reloaded the new AS 3.13.0, I got the log file in /esarchive/ even after a failure of one single SIM job inside a wrapper (with a different experiment though). I'm not totally sure it was exactly the same case than when I opened this issue but it seems to be. 

Many thanks @dbeltrankyl for the fix. I'm closing this issue as well.
